src/main/java/pl/kremiec/registrationapp/service/RestService.java

```diff
@@ -0,0 +1,46 @@
+package pl.kremiec.registrationapp.service;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.kremiec.registrationapp.model.Token;
+import pl.kremiec.registrationapp.model.User;
+import pl.kremiec.registrationapp.repo.TokenRepo;
+import pl.kremiec.registrationapp.repo.UserRepo;
+
+@Service
+public class RestService {
+
+    PasswordEncoder passwordEncoder;
+    UserRepo userRepo;
+    TokenRepo tokenRepo;
+    TokenService tokenService;
+
+    public RestService(PasswordEncoder passwordEncoder, UserRepo userRepo, TokenRepo tokenRepo, TokenService tokenService) {
+        this.passwordEncoder = passwordEncoder;
+        this.userRepo = userRepo;
+        this.tokenRepo = tokenRepo;
+        this.tokenService = tokenService;
+    }
+
+    public void saveUserToRepo(User user){
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        user.setRole("ROLE_USER");
+        userRepo.save(user);
+        tokenService.sendTokenToUser(user);
+    }
+
+    public void activateUser(String value){
+        Token byValue = tokenRepo.findByToken(value);
```

Comment:
How about:
```suggestion
        Optional.ofNullable(tokenRepo.findByToken(value)).ifPresentOrElse(
        token -> {
        User user = byValue.getUser();
        user.setEmailConfirmed(true);
        userRepo.save(user);
        }
        () -> {
        // return something that indicate that user was not updated, or throw specific exception
        }
        );  
```

And if'd asked me: I'm against throwing exception as a code execution control mechanism.